Sources/Environment.swift

```diff
@@ -33,3 +38,125 @@ public func stencilSwiftEnvironment() -> Environment {
 
   return Environment(extensions: [ext], templateClass: StencilSwiftTemplate.self)
 }
+
+extension Stencil.Extension {
+  // The following swiftlint annotation needs to be deleted once a swiftlint version including this PR
+  // https://github.com/realm/SwiftLint/pull/1725 is released.
+  // swiftlint:disable:next large_tuple
+  func registerFilterWithTwoArguments<T, A, B>(_ name: String, filter: @escaping (T, A, B) throws -> Any?) {
+    registerFilter(name) { (any, args) throws -> Any? in
+      guard let type = any as? T else { return any }
+      guard args.count == 2, let argA = args[0] as? A, let argB = args[1] as? B else {
+        throw TemplateSyntaxError("'\(name)' filter takes two arguments: \(A.self) and \(B.self)")
+      }
+      return try filter(type, argA, argB)
+    }
+  }
+
+  func registerFilterWithArguments<A>(_ name: String, filter: @escaping (Any?, A) throws -> Any?) {
+    registerFilter(name) { (any, args) throws -> Any? in
+      guard args.count == 1, let arg = args.first as? A else {
+        throw TemplateSyntaxError("'\(name)' filter takes a single \(A.self) argument")
+      }
+      return try filter(any, arg)
+    }
+  }
+
+  func registerBoolFilterWithArguments<U, A>(_ name: String, filter: @escaping (U, A) -> Bool) {
+    registerFilterWithArguments(name, filter: Filter.make(filter))
+    registerFilterWithArguments("!\(name)", filter: Filter.make({ !filter($0, $1) }))
+  }
+}
+
+private struct Filter<T> {
```

Comment:
> Agreed. And am I missing something, or is FilterOr never even used?

Exactly, I think I should've been much more careful with copy paste from [Sourcery](https://github.com/krzysztofzablocki/Sourcery). It's removed in 5f9819d.